### PR TITLE
Docs: add "how to share data"

### LINKED ISSUE #3998: Docs: How to share data

### LINKED ISSUE #4432: Docs: How to export AiiDA data for a publication

### PR DESCRIPTION
**Pull request by ltalirz**
fixes #4432 
fixes #3998 
partly addresses #4450

The "how to share data" section includes instructions both for dealing with AiiDA archives (e.g. for publishing AiiDA graphs alongside your publication) and for using the AiiDA REST API.

### CODE REVIEW

**Comment by sphuber** [comment]
@ltalirz there seems to be a problem with your fork when it comes to the Github Actions. With the other PR that was blocked, when I pushed to `aiidateam`, it ran without issues. @chrisjsewell also just opened a new PR and there all actions are running without issue. So I think it must be something with your fork, but not sure what. Maybe it has to do with the >150 branches that you have? Could be worth trying to delete all the stale ones you no longer need.

File: docs/source/howto/export.rst
Comment (giovannipizzi) [inline docs/source/howto/export.rst:50]: I would put MC Archive as the first one ;-)

File: docs/source/howto/export.rst
Comment (giovannipizzi) [inline docs/source/howto/export.rst:53]: I would move this comment earlier, around line 35, after the paragraph "If you want to make sure that the AiiDA export includes everything you intended"

**Review by giovannipizzi — COMMENTED** [review]
Two comments.

In addition, we're discussing with @sphuber about the number of top-level how-tos.
This indeed merits a section, but maybe we can make it a bit more general, not just on how to export for publication, but on how to export in general, and then keep the content of this PR for a subsection.

This is related to #3998, pinging @CasperWA to make sure we make a unified decision on the structure of the 'export/share' section

**Comment by chrisjsewell** [comment]
> @ltalirz there seems to be a problem with your fork when it comes to the Github Actions. 

@sphuber actually I think this is because of ff7b9e630967a4aece8f7dd75c052366920cd94e; so these tests don't run for docs only changes, but they are still set as required

**Comment by CasperWA** [comment]
> This is related to #3998, pinging @CasperWA to make sure we make a unified decision on the structure of the 'export/share' section

I'm setting this one up now.
Basically I am just pasting over the old import/export section and updating it accordingly, leaving out the technical sections concerning the JSON files.

@ltalirz so if you're concerned with overlaps, please take a look at the [old import/export section](https://aiida.readthedocs.io/projects/aiida-core/en/v1.1.0/import_export/main.html) to compare.

**Comment by ltalirz** [comment]
@giovannipizzi @CasperWA  I'm done with the "Howto share data" section for the moment, feel free to comment & review

Two notes:
 * Some archive-version-related details from `import_export/main.rst ` have not been moved to the howto - I've informed @ramirezfranciscof of exactly what those are so he includes them in the documentation of the archive format 
 * I still need to update the instructions for hosting your own REST API server - I will do this in a second step once this is merged.

File: docs/source/howto/share_data.rst
Comment (CasperWA) [inline docs/source/howto/share_data.rst:58]:
```suggestion
Publishing AiiDA archive files
```

File: docs/source/howto/share_data.rst
Comment (CasperWA) [inline docs/source/howto/share_data.rst:61]:
```suggestion
AiiDA archive files can be published on any research data repository, for example the `Materials Cloud Archive`_, `Zenodo`_, or the `Open Science Framework`_.
```

File: docs/source/howto/share_data.rst
Comment (CasperWA) [inline docs/source/howto/share_data.rst:62]: Please find-and-replace all of these.
```suggestion
When publishing AiiDA archive files on the `Materials Cloud Archive`_, you also get an interactive *EXPLORE* section, which allows peers to browse the AiiDA provenance graph directly in the browser.
```

File: docs/source/howto/share_data.rst
Comment (CasperWA) [inline docs/source/howto/share_data.rst:82]:
```suggestion
Node extras and comments have special modes for determining how to import them - for more details, see ``verdi import --help``.
```

File: docs/source/howto/share_data.rst
Comment (CasperWA) [inline docs/source/howto/share_data.rst:80]: Maybe not necessary, since the entries will not have anything directly to do with the Python representation of the same.
```suggestion
During import, AiiDA will avoid identifier collisions and node duplication based on UUIDs (and email comparisons for :py:class:~`aiida.orm.users.User` entries).
```

File: docs/source/howto/share_data.rst
Comment (CasperWA) [inline docs/source/howto/share_data.rst:156]:
```suggestion
```

File: docs/source/howto/share_data.rst
Comment (CasperWA) [inline docs/source/howto/share_data.rst:157]: Why mislead here and point to a MaterialsCloud REST API if you're showing the URL of something different?
Perhaps make it more clear that you're sending people to MaterialsCloud
```suggestion
Open the base URL of the REST API (by default: http://127.0.0.1:5000/api/v4, `example on MaterialsCloud <https://aiida-dev.materialscloud.org/2dstructures/api/v4/nodes>`_) in your web browser in order to get an overview of the available endpoints.
```

File: docs/source/howto/share_data.rst
Comment (CasperWA) [inline docs/source/howto/share_data.rst:161]:
```suggestion
```

File: docs/source/howto/share_data.rst
Comment (CasperWA) [inline docs/source/howto/share_data.rst:85]: Does this work with multi-line + in-line?

File: docs/source/howto/share_data.rst
Comment (CasperWA) [inline docs/source/howto/share_data.rst:170]:
```suggestion
```

File: docs/source/howto/share_data.rst
Comment (CasperWA) [inline docs/source/howto/share_data.rst:189]:
```suggestion
The path where you put the ``.wsgi`` file as well as its name are irrelevant as long as they are correctly referred to in the Apache configuration file, as shown later on.
```

File: docs/source/howto/share_data.rst
Comment (CasperWA) [inline docs/source/howto/share_data.rst:192]:
```suggestion
In ``rest.wsgi`` the only options you might want to change is ``catch_internal_server``.
```

File: docs/source/howto/share_data.rst
Comment (CasperWA) [inline docs/source/howto/share_data.rst:199]:
```suggestion
Apache usually creates multiple processes dynamically and with this configuration each process will handle both apps.
```

File: docs/source/howto/share_data.rst
Comment (CasperWA) [inline docs/source/howto/share_data.rst:205]:
```suggestion
Let us call the two apps for this example ``django`` and ``sqlalchemy``, matching with the chosen AiiDA profiles.
```

File: docs/source/howto/share_data.rst
Comment (CasperWA) [inline docs/source/howto/share_data.rst:224]:
```suggestion
            Require all granted
```

File: docs/source/howto/share_data.rst
Comment (CasperWA) [inline docs/source/howto/share_data.rst:236]:
```suggestion
Nevertheless, we suggest users who need finer tuning of the deployment setup to look into to the official documentation of `Apache <https://httpd.apache.org/>`_ and, more importantly, `WSGI <wsgi.readthedocs.io/>`_.
```

File: docs/source/howto/share_data.rst
Comment (CasperWA) [inline docs/source/howto/share_data.rst:247]:
```suggestion
The first (second) request will be handled by the app ``django`` (``sqlalchemy``), and will serve results fetched from the AiiDA profile ``django`` (``sqlalchemy``).
```

File: docs/source/howto/share_data.rst
Comment (CasperWA) [inline docs/source/howto/share_data.rst:248]:
```suggestion
Notice that we have not specified any port in the URLs since Apache listens conventionally to port 80, where any request lacking the port is automatically redirected (port 443 for HTTPS).
```

**Review by CasperWA — CHANGES_REQUESTED** [review]

File: docs/source/howto/share_data.rst
Comment (ltalirz) [reply docs/source/howto/share_data.rst:157]: This was a copy/paste error, will fix this

File: docs/source/howto/share_data.rst
Comment (ltalirz) [reply docs/source/howto/share_data.rst:85]: thanks, I checked and it works fine.

**Comment by ltalirz** [comment]
Thanks for the detailed read @CasperWA ! 
I believe I have addressed all your comments and suggestions - please have a second look.

File: docs/source/howto/share_data.rst
Comment (sphuber) [inline docs/source/howto/share_data.rst:118]: Please add a link here:
```
.. _how-to:data:serve:launch:
```
I'd like to reference it

File: docs/source/howto/share_data.rst
Comment (sphuber) [inline docs/source/howto/share_data.rst:152]: Please add a link here:
```
.. _how-to:data:serve:query:
```
I'd like to reference it

File: docs/source/howto/share_data.rst
Comment (ltalirz) [reply docs/source/howto/share_data.rst:118]: thanks, will do!

I just fixed the section name, it will be `how-to:share:serve:launch`

File: docs/source/howto/share_data.rst
Comment (ltalirz) [reply docs/source/howto/share_data.rst:154]: done!

File: docs/source/howto/share_data.rst
Comment (CasperWA) [inline docs/source/howto/share_data.rst:18]:
```suggestion
Tell AiiDA the **final results** you would like to be reproducible, and AiiDA will automatically include their entire provenance using the :ref:`topics:provenance:consistency:traversal-rules`.
```

File: docs/source/howto/share_data.rst
Comment (CasperWA) [inline docs/source/howto/share_data.rst:33]:
```suggestion
See ``verdi export create --help`` for ways to modify the traversal rules.
```

File: docs/source/howto/share_data.rst
Comment (CasperWA) [inline docs/source/howto/share_data.rst:37]:
```suggestion
    If you want to make sure the archive includes everything you intended, you can create a new profile and import it:
```

File: docs/source/howto/share_data.rst
Comment (CasperWA) [inline docs/source/howto/share_data.rst:44]:
```suggestion
        $ verdi quicksetup --profile test-export  # create new profile
        $ verdi --profile test-export import my-calculations.aiida

    Now use, e.g. the :py:class:`~aiida.orm.querybuilder.QueryBuilder` to query the database.
```

File: docs/source/howto/share_data.rst
Comment (CasperWA) [inline docs/source/howto/share_data.rst:46]: Again, please perform a search-and-replace for these cases.
```suggestion
Please remember to use **UUIDs** when pointing your colleagues to data *inside* an AiiDA archive file, since UUIDs are guaranteed to be universally unique (while PKs aren't).
```

File: docs/source/howto/share_data.rst
Comment (CasperWA) [inline docs/source/howto/share_data.rst:64]:
```suggestion
```

File: docs/source/howto/share_data.rst
Comment (CasperWA) [inline docs/source/howto/share_data.rst:87]:
```suggestion
    You can also use ``verdi export migrate`` to create updated archive files from existing archive files (or update them in place).
```

File: docs/source/howto/share_data.rst
Comment (CasperWA) [inline docs/source/howto/share_data.rst:145]:
```suggestion
```

**Review by CasperWA — CHANGES_REQUESTED** [review]

File: docs/source/howto/share_data.rst
Comment (CasperWA) [reply docs/source/howto/share_data.rst]: Did you perform a search-and-replace for these cases? Or are you sure all cases have been addressed?

Comment (ltalirz) [reply docs/source/howto/share_data.rst]: Ok, I'll do a search & replace on the AiiDA codebase

Comment (CasperWA) [reply docs/source/howto/share_data.rst]: I was thinking more within this document (and the documentation in general), but uhm, sure 😅 But maybe this can also wait for a separate PR?

Comment (sphuber) [reply docs/source/howto/share_data.rst]: I think we can do this once we add the new implementation. We will anyway have to rewrite parts of the documentation. Wouldn't hold up this PR for that

Comment (CasperWA) [reply docs/source/howto/share_data.rst]: Indeed - but it seems @ltalirz is too fast for us both - already went and did it 😮

**Review by CasperWA — APPROVED** [review]
Cheers @ltalirz !

I'll merge this in as two separate commits:
- Updating documentation, adding the new "How to share data" page and removing the old sections.
- Using the new naming convention for AiiDA archive files instead of AiiDA export archive files (or similar).

**Pull request merged**
